### ISSUE TITLE
Cas::getCurrentUser() Exception

### ISSUE THREAD

**Issue by rfiedler** [XavRsl/Cas#12]
Hi Xavier ! Can you help me ?

I will describe the context of what is happening;

1- I access first page os system and do login
![laravel1](https://cloud.githubusercontent.com/assets/1020057/5873276/be72dfc6-a2db-11e4-80cc-bc47ce9b65a3.jpg)

2- Login redirect to cas server and authentication is ok, return to application
![laravel2](https://cloud.githubusercontent.com/assets/1020057/5873287/da29509c-a2db-11e4-860e-e30acc4fbef1.jpg)

3- Cas send ticket via GET but raise exception when getCurrentUser is called
![laravel3](https://cloud.githubusercontent.com/assets/1020057/5873305/0d5c5b1c-a2dc-11e4-9aca-71e8ba8e97e3.jpg)

Route, Filter and Controller informations :

Routes
![laravel4](https://cloud.githubusercontent.com/assets/1020057/5873320/3066a496-a2dc-11e4-8a6b-13bf20f4b0b0.jpg)

Filters
![laravel5](https://cloud.githubusercontent.com/assets/1020057/5873324/341b349e-a2dc-11e4-8727-ad4eb0e02afb.jpg)

Controller
![laravel6](https://cloud.githubusercontent.com/assets/1020057/5873337/63f4b97e-a2dc-11e4-95d9-998d8c284e91.jpg)

Thank you for your attention

**Comment by XavRsl** [comment]
Hi,
It seems weird to have 'CASAuthServer' suffix in your ticket. I don't really know where that comes from...
Your code seams alright. And if it works with the Jasig package, it's even weirder... Do you see that 'CASAuthServer' string in your URL as well when you're using the Jasig package ?

Xavier

**Comment by jmandrade** [comment]
Same error here. Works OK without "ca", doesn't work with. I think "ufscCASAuthServer" is actually the server name (I get a different string there). 

In my case the error seems to happen on "Cas::authenticate();"

**Comment by XavRsl** [comment]
Is your application using https protocol ? If you're using ca to
authenticate the server, you will need to do it over https.

Xavier

2015-02-10 19:55 GMT+01:00 Jose Andrade <notifications@github.com>:

> Same error here. Works OK without "ca", doesn't work with. I think
> "ufscCASAuthServer" is actually the server name (I get a different string
> there).
>
> In my case the error seems to happen on "Cas::authenticate();"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/XavRsl/Cas/issues/12#issuecomment-73759460>.
>



-- 
Xavier

**Comment by jmandrade** [comment]
You're right! Thank you! - Problem fixed. It works OK on the production server with SSL enabled.
